### PR TITLE
Let Figure.savefig support filenames with spaces

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

`Figure.savefig()` doesn't work if filenames contain spaces.

Here is an minimal example to reproduce the issue:
```
import pygmt
fig = pygmt.Figure()
fig.basemap(projection="X10c/10c", frame=True, region=[0, 10, 0, 10])
fig.savefig("map 2.png")
```

This PR fixes the issue by adding double quotes around filenames. 

**NOTE:** Using filenames with spaces is NOT recommended. 
However, we have to support it, because sometimes filenames of some 
baseline images must contain spaces. 
See https://github.com/GenericMappingTools/pygmt/runs/2179626599 for an example failure.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by weiji14** [comment]
> **NOTE:** Using filenames with spaces is NOT recommended.
> However, we have to support it, because sometimes filenames of some
> baseline images must contain spaces.

See my comment at https://github.com/GenericMappingTools/pygmt/pull/804#discussion_r600953519 on how to fix the space issue. If the tests pass, do we still need this PR?

**Comment by seisman** [comment]
Nice fix in https://github.com/GenericMappingTools/pygmt/pull/804#discussion_r600968043.

I think this PR is still needed for people who want to use spaces. It's allowed in GMT (https://docs.generic-mapping-tools.org/dev/begin.html#examples).

File: pygmt/figure.py [hunk 1]
```diff
@@ -170,6 +170,9 @@ def psconvert(self, **kwargs):
         # Default cropping the figure to True
         if "A" not in kwargs:
             kwargs["A"] = ""
+        # allow for spaces in figure name
+        if kwargs.get("F"):
+            kwargs["F"] = f'"{kwargs.get("F")}"'
```
Comment (weiji14) [inline pygmt/figure.py:175]: > I think this PR is still needed for people who want to use spaces. It's allowed in GMT (https://docs.generic-mapping-tools.org/dev/begin.html#examples).

Yes, but this is just a patchy solution to the main underlying problem which is #247. I mitigated it in 4126c16160c67f8d2c405e7d5da56145adc989d4 for subplot's `title` parameter too, but I wonder if it's time we have a real, standardized solution that works to fix #247 entirely:

```suggestion
        kwargs["F"] = f'"{kwargs.get("F")}"' if kwargs.get("F") else None
```

Comment (seisman) [reply pygmt/figure.py]: subprocess provides a function [`list2cmdline`](https://github.com/python/cpython/blob/3.9/Lib/subprocess.py#L533) which may be useful. I think I tried it once. It works but may break backward-compatibility.

Comment (weiji14) [reply pygmt/figure.py]: Oo, that looks promising. How does it work, and why do you think it might break backward-compatibility?

Comment (seisman) [reply pygmt/figure.py]: I can't find my old testing codes. If I remember it correctly.

Here is the current behavior:
```
# 1st way: FAIL
fig.basemap(R=[0, 100, 0, 50], J="X10c/5c", B='x20+lCrust Age')
# 2nd way: SUCCEED
fig.basemap(R=[0, 100, 0, 50], J="X10c/5c", B='x20+l"Crust Age"')
# 3rd way: FAIL
fig.basemap(R=[0, 100, 0, 50], J="X10c/5c", B="x20+l'Crust Age'")
```
After using `list2cmdline`, all the three options work, but options 2 and 3 will have single or double quotes in the X labels.

Comment (weiji14) [reply pygmt/figure.py]: I see, but maybe we can leave the complicated frame/B option as it is now and only use `list2cmdline` for less complicated parameters, like `fig.text(text=...)` and `fig.savefig(fname=...)`? That itself would be a step in the right direction.

Comment (seisman) [reply pygmt/figure.py]: We need more investigation on the function and how it can be useful in our project.

**Comment by seisman** [comment]
@weiji14 Any other concerns?

**Comment by seisman** [comment]
Ping @GenericMappingTools/pygmt-contributors for comments if we should allow whitespace in figure names.

**Comment by core-man** [comment]
> Ping @GenericMappingTools/pygmt-contributors for comments if we should allow whitespace in figure names.

I think so and this PR is simple enough.

File: pygmt/tests/test_figure.py [hunk 2]
```diff
@@ -96,6 +96,18 @@ def test_figure_savefig_transparent():
...
+def test_figure_savefig_filename_with_spaces():
+    """
+    Check if savefig (or psconvert) supports filenames with spaces.
+    """
+    fig = Figure()
+    fig.basemap(region=[0, 1, 0, 1], projection="X1c/1c", frame=True)
+    fname = "test_figure_savefig filename with spaces.png"
+    fig.savefig(fname)
+    assert os.path.exists(fname)
+    os.remove(fname)
```
Comment (weiji14) [inline pygmt/tests/test_figure.py:108]: Probably better to use GMTTempfile if possible here.

**Review by weiji14 — COMMENTED** [review]
I'm still reluctant to allow spaces since it adds an if-clause to the code, but ok with supporting it if three or more people agree.

**Comment by willschlitzer** [comment]
> Ping @GenericMappingTools/pygmt-contributors for comments if we should allow whitespace in figure names.

I think it is good to allow this. I'm glad there is the disclaimer that spaces in filenames isn't recommended, but better to increase compatibility.

File: pygmt/tests/test_figure.py (hunk 2 above)
Comment (seisman) [reply pygmt/tests/test_figure.py]: Done in 4d86e7816cce4b6cf297278e56f20228ab976c4e.

**Review by meghanrjones — APPROVED** [review]

**Pull request merged**